analyze_impact.py

```diff
@@ -36,72 +45,105 @@ def get_linked_columns(
             raise NotFoundError('Column %s is not found' % c)
         column_uids.append(uid)
 
-    # print('table uid:', table_uid)
-    # print('column uid:', column_uids)
-
     # Now let's get all downstreams
-    lineage_data = client.execute(load_query('queries/lineage_analyze_impact.gql'), {
-        'primaryUid': table_uid,
-        'depthDownstream': 1000,
-        'depthUpstream': 0,
-        'popularity': [0, 4],
-        'biLastUsedDays': 90,
-        'allowedList': column_uids,
-    })
-
-    columns_to_ignore = set(column_name_to_uid.values()) - set(column_uids)
-    connected_columns = set()
-    for edge in lineage_data['lineage']['edges']:
-        if edge['sourceUid'] is None or edge['destinationUid'] is None:
-            continue # this is an "off-chart" edge
-
-        connected_columns.add(edge['sourceUid'])
-        connected_columns.add(edge['destinationUid'])
-
-    columns_per_table = defaultdict(set)
-    for col in lineage_data['lineage']['entities']:
-        uid = col.get('uid')
-        if uid is None:
-            continue  # that's not a column
-
-        if uid not in connected_columns:
-            continue
-
-        table = col['table']['prop']['path']
-        columns_per_table[table].add(col['prop']['name'])
-
-    return columns_per_table
-
-
-def main():
-    parser = argparse.ArgumentParser(description='Analyze downstream dependencies')
-    parser.add_argument('data_source_id', type=int)
-    parser.add_argument('full_table_name', type=str,
-                        help='db.schema.name format, case-sensitive')
-    parser.add_argument('columns', metavar='column', type=str, nargs='*',
-                        help='column name to trace; ommit to trace all columns')
+    lineage_data = execute_stored_query(
+        QUERIES / 'lineage_analyze_impact.gql',
+        primaryUid=table_uid,
+        depthDownstream=1000,
```

Comment:
Should we use a small value here for the default value?

Not sure who and why would have that level of depth, but even if we encounter a bug on our end, it's safer to have it lower.

default to 25 but allow up to 100?

Reply:
This is for impact analysis so we have to traverse the whole depth of the graph. The value of 1000 was used here before me; in data diffs UI the front end uses the value of 99.

We might want to create a separate endpoint which would ensure to travel the whole depth — but we haven't done that yet.

Reply:
I thought we'd help customers run this type of queries on their own? 🤔